src/actions/ImportActions.js

```diff
@@ -0,0 +1,60 @@
+import alt from 'utils/alt';
+import Import from 'utils/import';
+
+class ImportActions {
+  constructor() {
+    this.generateActions(
+      'importProgress',
+      'importSuccess',
+      'importFailure',
+      'detectTypeSuccess',
+      'detectTypeFailure',
+      'defaultStore',
+      'changeSpecial'
+    );
+  }
+
+  detectType({ file }) {
+    const reader = new FileReader();
+    reader.readAsText(file);
+    reader.onload = readedFile => {
+      const file = readedFile.target.result;
+      let importType = '';
+      const importTypes = Object.keys(Import);
+      for (let i = 0; i < importTypes.length; i++) {
+        if (Import[importTypes[i]].detect(file)) {
+          importType = importTypes[i];
+          break;
+        }
+      }
+      if (importType !== '') {
+        const special = Import[importType].needSpecial();
```

Comment:
I would stock the mandatory import `inputs` in the importer directly as an object. Not as a function needed to be called.